Assets/VRTK/Prefabs/SnapDropZone/VRTK_SnapDropZone.cs

```diff
@@ -316,6 +316,68 @@ public virtual VRTK_InteractableObject GetCurrentSnappedInteractableObject()
             return currentSnappedObject;
         }
 
+        /// <summary>
+        /// The Clone method returns the GameObject of the cloned snap drop zone
+        /// </summary>
+        /// <param name="position">Position of the cloned GameObject</param>
+        /// <returns>The GameObject of the clone</returns>
+        public virtual GameObject Clone(Vector3 position)
+        {
+            VRTK_SnapDropZone cloneSDZ = Instantiate(gameObject, position, transform.rotation).GetComponent<VRTK_SnapDropZone>();
+
+            for (int childID = 0; childID < cloneSDZ.transform.childCount; childID++)
+            {
+                Transform child = cloneSDZ.transform.GetChild(childID);
+                if (child.GetComponent<VRTK_InteractableObject>() != null)
+                    Destroy(child.gameObject);
+            }
+
+            if (isSnapped)
+            {
+                VRTK_InteractableObject currObject = currentSnappedObject;
+
+                //Get copy of Objects original state
+                Transform previousParent;
+                bool previousKinematic;
+                bool previousGrabbable;
+                currObject.GetPreviousState(out previousParent, out previousKinematic, out previousGrabbable);
+
+                GameObject clonedObject = null;
+                if (cloneNewOnUnsnap)
+                {
+                    clonedObject = Instantiate(objectToClone);
+                    clonedObject.SetActive(true);
+                }
+                else
+                {
+                    clonedObject = Instantiate(currObject.gameObject);
+                }
+
+                clonedObject.transform.position = cloneSDZ.transform.position;
+                cloneSDZ.ForceSnap(clonedObject);
+
+                StartCoroutine(OverridePreviousStateAtEndOfFrame(clonedObject.GetComponent<VRTK_InteractableObject>(), previousParent, previousKinematic, previousGrabbable));
```

Comment:
worth saving coroutine state into a variable, if the script is disabled you can then cancel the coroutine (or if this is called again and the coroutine is still running then you can cancel the previous one).

otherwise you'll be in a situation where multiple coroutines can be running and overriding each other. Or the script is disabled and the coroutine still fires.

None of that may be an issue, i've not read down far enough yet to check :)